File: src/main/java/top/potmot/jimmercodegen/config/DataSourceConfig.java
package top.potmot.jimmercodegen.config;

import com.alibaba.druid.pool.DruidDataSource;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

import javax.sql.DataSource;

@Configuration
public class DataSourceConfig {
    @Bean(name = "PrimaryDataSource")
    @Primary
    @ConfigurationProperties(prefix = "spring.datasource.primary")
    public DataSource getPrimaryDateSource() {
        return new DruidDataSource();
    }

    @Bean(name = "GenDataSource")
    @ConfigurationProperties(prefix = "spring.datasource.gen")
    public DataSource getGenDateSource() {
        return new DruidDataSource();
    }

}

File: src/main/kotlin/top/potmot/config/DataSourceConfig.kt
package top.potmot.config

import com.zaxxer.hikari.HikariDataSource
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import javax.sql.DataSource

@Configuration
class DataSourceConfig {
    @get:ConfigurationProperties(prefix = "spring.datasource.main")
    @get:Primary
    @get:Bean(name = ["MainDataSource"])
    val mainDateSource: DataSource
        get() = HikariDataSource()

    @get:ConfigurationProperties(prefix = "spring.datasource.gen")
    @get:Bean(name = ["GenDataSource"])
    val genDateSource: DataSource
        get() = HikariDataSource()
}
